### DISCUSSION TITLE
Help with implementing Preact.

### DISCUSSION THREAD

**Discussion by Julian1993ARG** [serenity-is/Serenity#6995] [Help] [answered]
Hello, can someone provide me with an example of implementing Preact? I see it installed in the packages, but I can't find an example of how to use it.

When i use:
/** @jsxImportSource preact */
I have this error, 'cannot be userd as a JSX component, its return type 'Element' is not a valid JSX element.

**Comment by volkanceylan** [comment]
Your project probably not using new jsx automatic runtime, so is still using jsx-dom

You should have these two lines in tsconfig.json:

    "jsx": "react-jsx",
    "jsxImportSource": "preact",

I think we have this in Serene:

    "jsx": "react",

Which means classic runtime, so you should change to what I mention above

**Reply by Julian1993ARG** [reply]
Yes, i have this config, but i would like to use preact Hooks [Preact](https://preactjs.com/guide/v10/hooks) but when i use it, give me this error.
```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '__H')
    at d (index.js:137:19)
    at s (index.js:165:20)
    at h (index.js:154:9)
    at SelectArea (SelectArea.tsx:10:31)
    at jsx (index.js:303:9)
    at RenderFilterBar (RenderFilterBar.tsx:9:9)
    at jsx (index.js:303:9)
    at n.<anonymous> (CalendarSolicitudesPage.tsx:15:13)
    at Generator.next (<anonymous>)
    at chunk-C2DPRSVE.js:1:1688
```

should i config something?

Right now, i am using [Rxjs](https://rxjs.dev/) to handle events because I know this library

**Comment by volkanceylan** [comment]
Also see latest commit in Serene, the package.json, tsconfig.json and LoginPage.tsx

**Comment by volkanceylan** [comment, marked as answer]
Sorry, cant offer support for third party libs here
